types/clevertap/index.d.ts

```diff
@@ -0,0 +1,168 @@
+// Type definitions for clevertap 1.2
+// Project: https://github.com/CleverTap/clevertap-node (Does not have to be to GitHub, but prefer linking to a source code repository rather than to a project website.)
+// Definitions by: Royson <https://github.com/droyson>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+type RequireAtLeastOne<T, Keys extends keyof T = keyof T> = Pick<T, Exclude<keyof T, Keys>> &
+    {
+        [K in Keys]-?: Required<Pick<T, K>> & Partial<Pick<T, Exclude<Keys, K>>>;
+    }[Keys];
+
+type Callback = (res: any) => void;
+
+type Identity = RequireAtLeastOne<
+    {
+        identity?: string;
+        FBID?: string;
+        GPID?: string;
+        objectId?: string;
+    },
+    'identity' | 'FBID' | 'GPID' | 'objectId'
+>;
+
+type ProfileUploadData = {
+    type: 'profile';
+    profileData: Record<string, any>;
+    ts?: number;
+} & Identity;
+
+type EventUploadData = {
+    type: 'event';
+    evtName: string;
+    evtData: Record<string, any>;
+    ts?: number;
+} & Identity;
+
+type UploadData = ProfileUploadData | EventUploadData;
+
+interface UploadOptions {
+    debug?: number;
+    batchSize?: number;
+}
+
+interface Query {
+    event_name: string;
+    props?: Array<Record<string, any>>;
+    from: number;
+    to: number;
+}
+
+interface QueryOptions {
+    debug?: number;
+    batchSize?: number;
+}
+
+type ProfileIdentity = RequireAtLeastOne<{
+    email?: string;
+    identity?: string;
+    objectId?: string;
+}>;
+
+type ProfileOptions = {
+    debug?: number;
+} & ProfileIdentity;
+
+interface CustomControlGroup {
+    type: 'custom';
+    name: string;
+}
+
+interface CampaignControlGroup {
+    type: 'campaign';
+    percentage: number;
+}
+
+interface TargetBasePayload {
+    name: string;
+    devices: string[];
+    when: string;
+    content: {
+        title: string;
+        body: string;
+        platform_specific?: Record<string, any>;
+        replacements?: Record<string, any>;
+    };
+    ttl?: number;
+    wzrk_cid?: string;
+    wzrk_bc?: number;
+    wzrk_bi?: string;
+    system_control_group_include?: boolean;
+    control_group?: CampaignControlGroup | CustomControlGroup;
+    wzrk_acts?: string;
+    respect_throttle?: boolean;
+    send_to_all_devices?: boolean;
+}
+
+type TargetSegmentPayload = {
+    segment: 'all';
+} & TargetBasePayload;
+
+type TargetWherePayload = {
+    where: {
+        event_name?: string;
+        from?: number;
+        to?: number;
+        common_profile_properties?: {
+            profile_fields: Array<Record<string, any>>;
+        };
+    };
+} & TargetBasePayload;
+
+type TargetCreatePayload = TargetWherePayload | TargetSegmentPayload;
+
+interface TargetOptions {
+    debug?: number;
+}
+
+interface clevertap {
```

Comment:
that's not common convention, is there a reason for lowercase?

Reply:
oh, so this is a class constructor func,
so this should be really `CleverTap`, shoudn't?

Reply:
Just trying to keep it consistent with the [example in README](https://github.com/CleverTap/clevertap-node#usage). Changing this to CleverTap would be confusing since the import all is also called CleverTap.

Reply:
One doesn't have to export interface 'CleverTap'. It's not directly used, one is supposed to  use factory method `init` to create instance (this is similar how we cope with problems in TS Node not exporting/exporting classes for symbols like 'Stream', one is supposed to use factory methods. Technically, there is no reason you export this. When used, TSC compiler will infer a type properly

Reply:
as a convention people are using e.g. 'CleverTapStatic', 'CleverTapApi', 'CleverTapImpl' when there is a possible class in symbol names